sympy/solvers/diophantine.py

```diff
@@ -875,9 +875,8 @@ def _diop_quadratic(var, coeff, t):
                     + (sqa*g*u**2 + D*u + sqa*F) // _c
 
                 for z0 in range(0, abs(_c)):
-                    if divisible(
-                            sqa*g*z0**2 + D*z0 + sqa*F,
-                            _c):
+                    if (divisible(sqa*g*z0**2 + D*z0 + sqa*F, _c) and
```

Comment:
Is it worth it to factor out `sqa*g*z0**2 + D*z0 + sqa*F` since we are running through a loop and having to recompute this `|_c|` times? Something like

```
_tmp = sqa*g*z0**2 + D*z0 + sqa*F
if (divisible(_tmp, _c) and divisible(_tmp*e*sqc - z0*_c, a*E - sqa*D*e*sqc):
```